src/drivers/cyphal/Subscribers/udral/Battery.hpp

```diff
@@ -163,7 +168,12 @@ class UavcanBmsSubscriber : public UavcanDynamicPortSubscriber
 
 			bat_status.capacity = parameters.design_capacity.coulomb / 3.6f; // Coulomb -> mAh
 			bat_status.cycle_count = parameters.cycle_count;
-			bat_status.serial_number = parameters.unique_id & 0xFFFF;
```

Comment:
@PetervdPerk May I ask you how
https://github.com/PX4/PX4-Autopilot/blob/0e32b155f3649c333f8f548a634421150bbbf7bd/src/drivers/cyphal/Subscribers/udral/Battery.hpp#L166
works? Isn't `parameters.unique_id` this array
https://github.com/PX4/PX4-Autopilot/blob/0e32b155f3649c333f8f548a634421150bbbf7bd/src/drivers/cyphal/NodeManager.hpp#L64
and hence the line takes the first 16 bits of the array pointer? 🤔

My suggestion now is to take the
> 128-bit long sequence that is likely to be globally unique per node

and convert it to a human-readable hexadecimal string for the battery serial number. Maybe you have more context if that's the right way to do it.